### PR TITLE
Make Celestia RPC timeout configurable

### LINKED ISSUE #478: Allow to set `request_timeout` for `CelestiaAdapter`
In order to handle slow responses from celestia node, request timeout needs to be configurable from config

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
Make the Celestia-node RPC timeout configurable

## Linked Issues
- Fixes #478 

## Testing
Unit testing this change is difficult because the `jsonrpsee::HttpClient` doesn't expose a getter for its current `timeout` value - so, this PR leaves the change untested.

## Docs
The example `rollup_config.toml` and its documentation have been updated.

### CODE REVIEW

**Review by citizen-stig — APPROVED** [review]

File: examples/demo-rollup/rollup_config.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ celestia_rpc_auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJw
 celestia_rpc_address = "http://127.0.0.1:26658"
 # The largest response the rollup will accept from the Celestia node. Defaults to 100 MB
 max_celestia_response_body_size = 104_857_600
+# The maximum time to wait for a response to an RPC query against Celestia node. Defaults to 30 seconds.
+celestia_rpc_timeout_seconds = 30
```
Comment (preston-evans98) [inline examples/demo-rollup/rollup_config.toml:13]: @citizen-stig should we update this config as well?

File: examples/demo-rollup/benches/rollup_config.toml [hunk 2]
```diff
@@ -9,6 +9,8 @@ celestia_rpc_auth_token = "MY.SECRET.TOKEN"
 celestia_rpc_address = "http://localhost:11111/"
 # The largest response the rollup will accept from the Celestia node. Defaults to 100 MB
 max_celestia_response_body_size = 104_857_600
+# The maximum time to wait for a response to an RPC query against Celestia node. Defaults to 30 seconds.
+celestia_rpc_timeout_seconds = 30
```
Comment (preston-evans98) [inline examples/demo-rollup/benches/rollup_config.toml:13]: @dubbelosix, it looks to me like the config for the benches is identical to the one for the demo. Am I missing something? If not, should we think about deduplicating?

Comment (preston-evans98) [reply examples/demo-rollup/benches/rollup_config.toml:13]: Synced offline - we'll probably remove this shortly. Not going to block this PR on it tho!

File: examples/demo-rollup/rollup_config.toml (hunk 1 above)
Comment (preston-evans98) [reply examples/demo-rollup/rollup_config.toml]: Synced with @citizen-stig - we'll bump this to 60 seconds.

**Pull request merged**